### PR TITLE
Team forge

### PR DESCRIPTION
**Pull request by mlsteele**
A start on forging team sigchains.

### CODE REVIEW

File: src/teamforge.iced [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+      merkle_triples: {} # map from keys with a dash "LeafID-HashMeta" to MerkleTriple's
+      expect: @chain.expect
+    @_link_id_gen = new LinkIDGen
+
+    # This is the user who is loading teams.
+    # They are also the default for forging actions when no one else is specified.
+    @default_user_label = "herb"
+
+    for user_label, user_desc of @chain.users
+      await user = new User {forge:@, label: user_label}, esc defer()
```
Comment (maxtaco) [inline src/teamforge.iced]: as a convention I've adopted, I never make constructors async. Can you do something like this instead?

```iced
class User
   @make : ({forge, label}, cb) ->
     user = new User { forge, label }
      # blah blah
      cb null, user

await User.make { forge : @, label : user_label }, esc defer user
```

File: src/teamforge.iced [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+      @users[user_label] = user
+      await user.apply @out, esc defer()
+
+      @out.merkle_triples["#{@out.users[user_label].uid}-#{@_hash_meta 1000}"] =
+        seqno: 1
+        id: @out.users[user_label].link_map[1] # link_id
+
+    for label, team_desc of @chain.teams
+      await team = new Team {forge:@, label}, esc defer()
```
Comment (maxtaco) [inline src/teamforge.iced]: same comment as above

File: src/teamforge.iced [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    base = "cd".repeat 32
+    num_str = number.toString()
+    (base[...-num_str.length] + num_str)
+
+  #-------------------
+
+  _gen_key : (typ, cb) ->
+    switch typ
+      when 'eddsa'
+        await kbpgp.kb.KeyManager.generate {}, cb
```
Comment (maxtaco) [inline src/teamforge.iced]: no `await` here

**Comment by mlsteele** [comment]
I really like the `@` syntax in the arguments and constructor, so how about this `make + _init`?

File: src/teamforge.iced [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+    base = "ef".repeat 32
+    num_str = (@n++).toString()
+    (base[...-num_str.length] + num_str)
+
+#===================================================
+
+class User
+  @make : (args, cb) ->
+    obj = new User
+    obj._init args, (err) -> cb err, obj
```
Comment (maxtaco) [inline src/teamforge.iced]: this is fine with me, though I personally would do something like:

```iced
@make : (args, cb) ->
    obj = new User args
    obj._init {}, (err) -> cb err, obj
```

**Pull request merged**